src/ui/views/ViewMobileOnboarding.qml

```diff
@@ -122,18 +122,62 @@ VPNFlickable {
                             }
                         }
                     }
-                    onClicked: {
+
+                    Component.onCompleted: {
+                        currentPanelValues._panelTitleText = headline;
+                        currentPanelValues._panelDescriptionText = subtitle;
+                        updatePanel.start();
+                    }
+
+                    function goToNextSlide() {
                         if (swipeView.currentIndex < onboardingModel.count - 1) {
                             swipeView.currentIndex += 1;
                         } else {
                             swipeView.currentIndex = 0;
                         }
                     }
-                    Component.onCompleted: {
-                        currentPanelValues._panelTitleText = headline;
-                        currentPanelValues._panelDescriptionText = subtitle;
-                        updatePanel.start();
+                    function goToPreviousSlide() {
+                        if (swipeView.currentIndex > 0) {
+                            swipeView.currentIndex -= 1;
+                        } else {
+                            swipeView.currentIndex = onboardingModel.count - 1;
+                        }
                     }
+
+                    MouseArea {
+                        property int previousMouseX: 0
+                        property int swipeDirection: 0
+
+                        anchors.fill: parent
+
+                        onPressed: {
+                            previousMouseX = parseInt(mouseX);
+                            swipeDirection = 0;
+                        }
+
+                        onReleased: {
+                            if (swipeDirection < 0) {
+                                goToPreviousSlide();
+                            } else if (swipeDirection > 0) {
+                                goToNextSlide();
+                            } else {
+                                goToNextSlide();
+                            }
+                        }
+
+                        onPositionChanged: {
+                            const parsedMouseX = parseInt(mouseX);
+                            if (previousMouseX > parsedMouseX) {
```

Comment:
This makes the area extremely reactive. We need to test what happens with small clicks. If it's too much, maybe we should have a bit of tolerance